File: examples/dekita/.claude/hooks/session-log-compressor.py
#!/usr/bin/env python3
"""セッション終了時にローテート済みログを圧縮。

Why:
    ログファイルのローテート後、古いファイル（.log.1, .log.2等）が
    ディスクを圧迫する。gzip圧縮してストレージを節約する。

What:
    - セッション終了時（Stop）に発火
    - execution/とmetrics/のローテート済みログを検索
    - .log.N形式のファイルをgzip圧縮
    - 圧縮した件数をログに記録

Remarks:
    - 非ブロック型（Stopフック）
    - ローテーション自体はcommon.pyが担当
    - 既に圧縮済み（.gz）のファイルはスキップ

Changelog:
    - silenvx/dekita#710: フック追加
"""

import json
import sys
from pathlib import Path

# Add hooks directory to path for imports
sys.path.insert(0, str(Path(__file__).parent))

from common import EXECUTION_LOG_DIR, METRICS_LOG_DIR
from lib.execution import compress_rotated_logs, log_hook_execution
from lib.session import parse_hook_input


def main():
    """Compress rotated logs at session end."""
    # Read hook input
    hook_input = parse_hook_input()

    # Skip if Stop hook is already active (prevent infinite loop)
    if hook_input.get("stop_hook_active"):
        print(json.dumps({"decision": "approve"}))
        return

    # Compress rotated logs in both directories
    total_compressed = 0
    total_compressed += compress_rotated_logs(EXECUTION_LOG_DIR)
    total_compressed += compress_rotated_logs(METRICS_LOG_DIR)

    # Log the result
    log_hook_execution(
        "session-log-compressor",
        "approve",
        f"Compressed {total_compressed} rotated log file(s)",
        {"compressed_count": total_compressed},
    )

    # Always approve - don't block session end
    print(json.dumps({"decision": "approve"}))


if __name__ == "__main__":
    main()

File: examples/dekita/.claude/hooks/handlers/session_log_compressor.ts
#!/usr/bin/env bun
/**
 * セッション終了時にローテート済みログを圧縮。
 *
 * Why:
 *   ログファイルのローテート後、古いファイル（.log.1, .log.2等）が
 *   ディスクを圧迫する。gzip圧縮してストレージを節約する。
 *
 * What:
 *   - セッション終了時（Stop）に発火
 *   - execution/とmetrics/のローテート済みログを検索
 *   - .log.N形式のファイルをgzip圧縮
 *   - 圧縮した件数をログに記録
 *
 * Remarks:
 *   - 非ブロック型（Stopフック）
 *   - ローテーション自体はcommon.pyが担当
 *   - 既に圧縮済み（.gz）のファイルはスキップ
 *
 * Changelog:
 *   - silenvx/dekita#710: フック追加（Python）
 *   - silenvx/dekita#3148: TypeScriptに移行
 */

import { EXECUTION_LOG_DIR, METRICS_LOG_DIR } from "../lib/common";
import { compressRotatedLogs, logHookExecution } from "../lib/logging";
import { parseHookInput } from "../lib/session";

const HOOK_NAME = "session-log-compressor";

// Note: EXECUTION_LOG_DIR and METRICS_LOG_DIR are already absolute paths from lib/common

async function main(): Promise<void> {
  // Read hook input
  const hookInput = await parseHookInput();
  const sessionId = hookInput?.session_id;

  // Skip if Stop hook is already active (prevent infinite loop)
  if (hookInput?.stop_hook_active) {
    console.log(JSON.stringify({}));
    return;
  }

  // Compress rotated logs in both directories
  let totalCompressed = 0;
  totalCompressed += await compressRotatedLogs(EXECUTION_LOG_DIR);
  totalCompressed += await compressRotatedLogs(METRICS_LOG_DIR);

  // Log the result
  await logHookExecution(
    HOOK_NAME,
    "approve",
    `Compressed ${totalCompressed} rotated log file(s)`,
    { compressed_count: totalCompressed },
    { sessionId },
  );

  // Always approve - don't block session end
  console.log(JSON.stringify({}));
}

if (import.meta.main) {
  main().catch((error) => {
    console.error(`[${HOOK_NAME}] Unexpected error:`, error);
    console.log(JSON.stringify({}));
    process.exit(0); // Don't block on error
  });
}
